core/src/main/scala/org/apache/spark/resource/ExecutorResourceRequests.scala

```diff
@@ -0,0 +1,161 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.resource
+
+import scala.collection.mutable
+
+import org.apache.spark.network.util.JavaUtils
+import org.apache.spark.resource.ResourceProfile._
+
+/**
+ * A set of Executor resource requests. This is used in conjunction with the ResourceProfile to
+ * programmatically specify the resources needed for an RDD that will be applied at the
+ * stage level.
+ *
+ * This api is currently private until the rest of the pieces are in place and then it
+ * will become public.
+ */
+private[spark] class ExecutorResourceRequests() extends Serializable {
+
+  private val _executorResources = new mutable.HashMap[String, ExecutorResourceRequest]()
+
+  def requests: Map[String, ExecutorResourceRequest] = _executorResources.toMap
+
+  /**
+   * Specify heap memory.
+   *
+   * @param amount Amount of memory.
+   * @param units Units of the amount. For things like Memory, default is no units,
+   *              only byte types (b, mb, gb, etc) are currently supported.
+   */
+  def memory(amount: Long, units: String): this.type = {
+    val rr = new ExecutorResourceRequest(MEMORY, amount, units)
```

Comment:
I was thinking that you would convert this into bytes, and then `ExecutorResourceRequest` wouldn't have any units, so comparisons would be easier later on.  (don't feel that strongly about it, though)

Reply:
Sure, I'll remove and then assume if we are comparing ResourceProfiles the units specified didn't matter. I was thinking of keeping so we didn't lose the original units the user passed in.  For instance, if someone created a profile with say 2G, when they looked at it, it would specifically say 2G, not 2048M (or equivalent bytes).  Its really not a big deal though. If we find some reason to really need it we can add it back

Reply:
I think we can store in MiB instead of bytes because that is what most of the memory configs do.

Reply:
sure, MiB makes sense